src/tests/benchmarks.cpp

```diff
@@ -391,6 +391,37 @@ TEST_CASE("with_latest_from")
     };
 }
 
+TEST_CASE("combine_latest")
+{
+    BENCHMARK_ADVANCED("combine_latest construction from observable via dot + subscribe")(Catch::Benchmark::Chronometer meter)
+    {
+        const auto obs_1 = rpp::source::just(1, 2, 3);
+        const auto obs_2 = rpp::source::just(4, 5, 6);
+        auto sub = rpp::specific_subscriber{[](const std::tuple<int, int>) {}};
+
+        meter.measure([&]
+        {
+            return obs_1.combine_latest(obs_2).subscribe(sub);
```

Comment:
You need vector of subs there. Because `measure` can invoke lambda multple times. In this case first launch "completes" sub and then you measure nothing (due to sub unsubscribed). Re-use code with creation of vector of subs from above

Reply:
Oh I see. This explains why you have `create()` upstream that never ends.